### ISSUE TITLE
Failed to load plugins.

### ISSUE THREAD

**Issue by heit0r** [robbert-vdh/yabridge#159]
### Thanks for giving yabridge a shot!

- [X] I read through both the troubleshooting and the known issues sections, and my issue wasn't listed there

### Problem description

I couldn't load VST3 plugins on Ardour6. I got many of the same notification: "Failed to start Wine plugin host".

The output of `$ yabridge-host-32.exe`shows:
yabridge host version 3.7.0 (32-bit compatibility mode)
Usage: yabridge-host-32.exe <plugin_type> <plugin_location> <endpoint_base_directory> <parent_pid>

Which means that everything seems to be fine with 32-bit plugins (?)

but the output of `$ yabridge-host.exe` shows:
wine client error:0: version mismatch 733/740.
Your wineserver binary was not upgraded correctly,
or you have an older one somewhere in your PATH.
Or maybe the wrong wineserver is still running?
O aplicativo não pôde ser iniciado ou não existe aplicativo associado ao arquivo
 especificado.
ShellExecuteEx falhou: Arquivo não encontrado.

Which means that wine may not be updated correctly? 

### What did you expect to happen?

Ardour6 to load everything.

### What actually happened?

Never worked. Ardour6 Plugin manager marked the plugin as 'Error" followed by the output:

VST3 module-path '/home/mojo/.vst3/yabridge/BUSTERse.vst3/Contents/x86_64-linux/BUSTERse.so'
[Info]: Scanning: /home/mojo/.vst3/yabridge/BUSTERse.vst3
15:33:34 [BUSTERse-59AKPqtx] Initializing yabridge version 3.7.0
15:33:34 [BUSTERse-59AKPqtx] host:          '/usr/bin/yabridge-host.exe'
15:33:34 [BUSTERse-59AKPqtx] plugin:        '/home/mojo/Audio/AO Plugins/BUSTERse.vst3'
15:33:34 [BUSTERse-59AKPqtx] plugin type:   'VST3'
15:33:34 [BUSTERse-59AKPqtx] realtime:      'yes'
15:33:34 [BUSTERse-59AKPqtx] sockets:       '/run/user/1000/yabridge-BUSTERse-59AKPqtx'
15:33:34 [BUSTERse-59AKPqtx] wine prefix:   '<default>'
15:33:34 [BUSTERse-59AKPqtx] wine version:  '6.19'
15:33:34 [BUSTERse-59AKPqtx] 
15:33:34 [BUSTERse-59AKPqtx] config from:   '<defaults>'
15:33:34 [BUSTERse-59AKPqtx] hosting mode:  'individually, 64-bit'
15:33:34 [BUSTERse-59AKPqtx] other options: '<none>'
15:33:34 [BUSTERse-59AKPqtx] 
15:33:34 [BUSTERse-59AKPqtx] Enabled features:
15:33:34 [BUSTERse-59AKPqtx] - bitbridge support
15:33:34 [BUSTERse-59AKPqtx] - VST3 support
15:33:34 [BUSTERse-59AKPqtx] 
15:33:34 [BUSTERse-59AKPqtx] [Wine STDERR] wine client error:0: version mismatch 733/739.
15:33:34 [BUSTERse-59AKPqtx] [Wine STDERR] Your wineserver binary was not upgraded correctly,
15:33:34 [BUSTERse-59AKPqtx] [Wine STDERR] or you have an older one somewhere in your PATH.
15:33:34 [BUSTERse-59AKPqtx] [Wine STDERR] Or maybe the wrong wineserver is still running?
15:33:34 [BUSTERse-59AKPqtx] [Wine STDOUT] O aplicativo no pde ser iniciado ou no existe aplicativo associado ao arquivo especificado.
15:33:34 [BUSTERse-59AKPqtx] [Wine STDOUT] ShellExecuteEx falhou: Arquivo no encontrado.
15:33:34 [BUSTERse-59AKPqtx] [Wine STDOUT] 
15:33:34 [BUSTERse-59AKPqtx] The Wine host process has exited unexpectedly. Check the output above for more information.
terminate called without an active exception
Error: signal 6
 ---8<---
  /usr/lib/libc.so.6(+0x3cda0) [0x7f2eba31bda0]
  gsignal
  abort
  /usr/lib/libstdc++.so.6(+0x99802) [0x7f2eba582802]
  /usr/lib/libstdc++.so.6(+0xa5c8a) [0x7f2eba58ec8a]
  /usr/lib/libstdc++.so.6(+0xa5cf7) [0x7f2eba58ecf7]
  /home/mojo/.vst3/yabridge/BUSTERse.vst3/Contents/x86_64-linux/BUSTERse.so(+0xfdf6a) [0x7f2eb6f4af6a]
  /home/mojo/.vst3/yabridge/BUSTERse.vst3/Contents/x86_64-linux/BUSTERse.so(+0x16dd92) [0x7f2eb6fbad92]
  /home/mojo/.vst3/yabridge/BUSTERse.vst3/Contents/x86_64-linux/BUSTERse.so(+0x16dcd1) [0x7f2eb6fbacd1]
  /home/mojo/.vst3/yabridge/BUSTERse.vst3/Contents/x86_64-linux/BUSTERse.so(+0x16dc19) [0x7f2eb6fbac19]
  /home/mojo/.vst3/yabridge/BUSTERse.vst3/Contents/x86_64-linux/BUSTERse.so(+0x16da52) [0x7f2eb6fbaa52]
  /home/mojo/.vst3/yabridge/BUSTERse.vst3/Contents/x86_64-linux/BUSTERse.so(+0x16cdb6) [0x7f2eb6fb9db6]
  /usr/lib/libstdc++.so.6(+0xd33c4) [0x7f2eba5bc3c4]
 --->8---
Scan Failed.

### Operating system

Linux mojo-PC 5.15.7-1-rt23-MANJARO #1 SMP PREEMPT_RT Fri Dec 17 19:45:21 CET 2021 x86_64 GNU/Linux

### How did you install yabridge?

by 'pacman -S yabridge yabridgectl'

### yabridge version

yabridge-3.7.0-6

### yabridgectl version

yabridgectl 3.7.0

### Wine version

wine-6.19. But I installed wine-staging 7.0.

### Plugin

BUSTERse (and others from analog obsession)

### Plugin type

VST3

### Plugin architecture

both 64-bit and 32-bit

### Host

Ardour6

### Debug log

I got no /tmp/yabridge.log. Probably because I rebooted the machine.

### Anything else?

I installed wine-staging 7.0, but the output of `wine --version` is wine-6.19. And I think this is why this line "wine client error:0: version mismatch 733/739" appears. I actually went over a bunch of forums about this problem, but didn't found any way to fix this. I think this is not a problem with yabridge itself, but with wine. I killed every wine process and reinstalled everything many times but nothing seems to work. Help, please. Thank you for your attention.

The output of `pacman -Qi wine` :

Nome                 : wine-staging
Versão               : 7.0rc2-1
Descrição            : A compatibility layer for running Windows programs -
                       Staging branch
Arquitetura          : x86_64
URL                  : https://www.wine-staging.com
Licenças             : LGPL
Grupos               : Nenhum
Provê                : wine=7.0rc2  wine-wow64=7.0rc2
Depende de           : attr  lib32-attr  fontconfig  lib32-fontconfig  lcms2
                       lib32-lcms2  libxml2  lib32-libxml2  libxcursor
                       lib32-libxcursor  libxrandr  lib32-libxrandr  libxdamage
                       lib32-libxdamage  libxi  lib32-libxi  gettext
                       lib32-gettext  freetype2  lib32-freetype2  glu
                       lib32-glu  libsm  lib32-libsm  gcc-libs  lib32-gcc-libs
                       libpcap  lib32-libpcap  faudio  lib32-faudio
                       desktop-file-utils
Depend. opcionais    : giflib [instalado]
                       lib32-giflib
                       libpng [instalado]
                       lib32-libpng [instalado]
                       libldap [instalado]
                       lib32-libldap [instalado]
                       gnutls [instalado]
                       lib32-gnutls
                       mpg123 [instalado]
                       lib32-mpg123 [instalado]
                       openal [instalado]
                       lib32-openal
                       v4l-utils [instalado]
                       lib32-v4l-utils
                       libpulse [instalado]
                       lib32-libpulse [instalado]
                       alsa-plugins [instalado]
                       lib32-alsa-plugins [instalado]
                       alsa-lib [instalado]
                       lib32-alsa-lib [instalado]
                       libjpeg-turbo [instalado]
                       lib32-libjpeg-turbo [instalado]
                       libxcomposite [instalado]
                       lib32-libxcomposite
                       libxinerama [instalado]
                       lib32-libxinerama
                       opencl-icd-loader
                       lib32-opencl-icd-loader
                       libxslt [instalado]
                       lib32-libxslt
                       libva [instalado]
                       lib32-libva [instalado]
                       gtk3 [instalado]
                       lib32-gtk3
                       gst-plugins-base-libs [instalado]
                       lib32-gst-plugins-base-libs
                       vulkan-icd-loader [instalado]
                       lib32-vulkan-icd-loader [instalado]
                       sdl2 [instalado]
                       lib32-sdl2 [instalado]
                       vkd3d [instalado]
                       lib32-vkd3d
                       sane [instalado]
                       libgphoto2 [instalado]
                       gsm [instalado]
                       ffmpeg [instalado]
                       cups [instalado]
                       samba [instalado]
                       dosbox
Necessário para      : winetricks  yabridge
Opcional para        : Nenhum
Conflita com         : wine  wine-wow64
Substitui            : Nenhum
Tamanho instalado    : 497,94 MiB
Empacotador          : Felix Yan <felixonmars@archlinux.org>
Data da compilação   : qui 30 dez 2021 16:33:50
Data de instalação   : ter 18 jan 2022 15:36:16
Motivo da instalação : Instalado explicitamente
Script de instalação : Sim
Validado por         : Assinatura

**Comment by robbert-vdh** [comment]
You still have a wineserver running for an older verison of Wine. Run `wineserver -k` to kill it.

**Comment by heit0r** [comment]
Thank you for your reply. I ran this command as normal user and superuser and rebooted the machine, but the problem remains.

An example:

VST3 module-path '/home/mojo/.vst3/yabridge/BlackVibe.vst3/Contents/x86_64-linux/BlackVibe.so'
[Info]: Scanning: /home/mojo/.vst3/yabridge/BlackVibe.vst3
18:31:24 [BlackVibe-9BGQSjqr] Initializing yabridge version 3.7.0
18:31:24 [BlackVibe-9BGQSjqr] host:          '/usr/bin/yabridge-host.exe'
18:31:24 [BlackVibe-9BGQSjqr] plugin:        '/home/mojo/Audio/AO Plugins/BlackVibe.vst3'
18:31:24 [BlackVibe-9BGQSjqr] plugin type:   'VST3'
18:31:24 [BlackVibe-9BGQSjqr] realtime:      'yes'
18:31:24 [BlackVibe-9BGQSjqr] sockets:       '/run/user/1000/yabridge-BlackVibe-9BGQSjqr'
18:31:24 [BlackVibe-9BGQSjqr] wine prefix:   '<default>'
18:31:24 [BlackVibe-9BGQSjqr] wine version:  '6.19'
18:31:24 [BlackVibe-9BGQSjqr] 
18:31:24 [BlackVibe-9BGQSjqr] config from:   '<defaults>'
18:31:24 [BlackVibe-9BGQSjqr] hosting mode:  'individually, 64-bit'
18:31:24 [BlackVibe-9BGQSjqr] other options: '<none>'
18:31:24 [BlackVibe-9BGQSjqr] 
18:31:24 [BlackVibe-9BGQSjqr] Enabled features:
18:31:24 [BlackVibe-9BGQSjqr] - bitbridge support
18:31:24 [BlackVibe-9BGQSjqr] - VST3 support
18:31:24 [BlackVibe-9BGQSjqr] 
18:31:25 [BlackVibe-9BGQSjqr] [Wine STDERR] wine client error:0: version mismatch 733/740.
18:31:25 [BlackVibe-9BGQSjqr] [Wine STDERR] Your wineserver binary was not upgraded correctly,
18:31:25 [BlackVibe-9BGQSjqr] [Wine STDERR] or you have an older one somewhere in your PATH.
18:31:25 [BlackVibe-9BGQSjqr] [Wine STDERR] Or maybe the wrong wineserver is still running?
18:31:25 [BlackVibe-9BGQSjqr] [Wine STDOUT] O aplicativo no pde ser iniciado ou no existe aplicativo associado ao arquivo especificado.
18:31:25 [BlackVibe-9BGQSjqr] [Wine STDOUT] ShellExecuteEx falhou: Arquivo no encontrado.
18:31:25 [BlackVibe-9BGQSjqr] [Wine STDOUT] 
18:31:25 [BlackVibe-9BGQSjqr] The Wine host process has exited unexpectedly. Check the output above for more information.
terminate called without an active exception
Error: signal 6
 ---8<---
  /usr/lib/libc.so.6(+0x3cda0) [0x7f885bd09da0]
  gsignal
  abort
  /usr/lib/libstdc++.so.6(+0x99802) [0x7f885bf70802]
  /usr/lib/libstdc++.so.6(+0xa5c8a) [0x7f885bf7cc8a]
  /usr/lib/libstdc++.so.6(+0xa5cf7) [0x7f885bf7ccf7]
  /home/mojo/.vst3/yabridge/BlackVibe.vst3/Contents/x86_64-linux/BlackVibe.so(+0xfdf6a) [0x7f8858938f6a]
  /home/mojo/.vst3/yabridge/BlackVibe.vst3/Contents/x86_64-linux/BlackVibe.so(+0x16dd92) [0x7f88589a8d92]
  /home/mojo/.vst3/yabridge/BlackVibe.vst3/Contents/x86_64-linux/BlackVibe.so(+0x16dcd1) [0x7f88589a8cd1]
  /home/mojo/.vst3/yabridge/BlackVibe.vst3/Contents/x86_64-linux/BlackVibe.so(+0x16dc19) [0x7f88589a8c19]
  /home/mojo/.vst3/yabridge/BlackVibe.vst3/Contents/x86_64-linux/BlackVibe.so(+0x16da52) [0x7f88589a8a52]
  /home/mojo/.vst3/yabridge/BlackVibe.vst3/Contents/x86_64-linux/BlackVibe.so(+0x16cdb6) [0x7f88589a7db6]
  /usr/lib/libstdc++.so.6(+0xd33c4) [0x7f885bfaa3c4]
 --->8---
Scan Failed.

**Comment by robbert-vdh** [comment]
Well you're somehow running using another Wine version in the prefix at the same time. Running `shopt -s globstar; fuser -v /tmp/.wine**/*` in a bash shell will point you towards the currently running `wineserver` processes.

**Comment by heit0r** [comment]
Thanks again, but nothing gives output:


![2022-01-18_19-09](https://user-images.githubusercontent.com/38054654/150026607-08423e14-c8b7-4202-8fd9-7eb5aaf4d45d.png)

Bashtop doesn't filter any `wine` process

![2022-01-18_19-10](https://user-images.githubusercontent.com/38054654/150026778-8d6552c6-a5ee-486e-aea4-b3fbbbb8576d.png)

**Comment by robbert-vdh** [comment]
I have no idea what could be causing this then, I've never seen that Wine error show up spuriously. You're running an older version of Wine, have you tried updating to the latest Wine Staging version yet? And just as a sanity check, what do the following commands print when run from a terminal? ```bash
echo $WINELOADER
which wine wine64 wine64-preloader wineserver
```

And can you run `winecfg` from a terminal? Does that print the same error? (it should) You may need to ask around on the Wine forums or on IRC.

**Comment by heit0r** [comment]
I got these outputs:

[mojo-PC mojo]# echo $WINELOADER
which wine wine64 wine64-preloader wineserver

/usr/local/bin/wine
/usr/bin/wine64
/usr/bin/wine64-preloader
/usr/local/bin/wineserver
[mojo-PC mojo]# winecfg
wine: '/root/.wine' is a 64-bit installation, it cannot be used with a 32-bit wineserver.
[mojo-PC mojo]# e
exit
[mojo@mojo-PC ~]$ winecfg

Running `winecfg`from non-root user pops up the winecfg GUI that shows wine as 6.19 again. 

![2022-01-18_19-44](https://user-images.githubusercontent.com/38054654/150030678-2f27a4b3-51e6-4415-809c-d591eb47f888.png)



I'm going to foward this issue to the wine team and meanwhile I'm asking you to do not close this issue here because I will post how I solved this, if I can.

Thanks a lot.

**Comment by robbert-vdh** [comment]
That ` /usr/local/bin/wineserver` is definitely the problem here. Why do you have a partial Wine installation in `/usr/local`? Wine's using the wineserver from `/usr/local/bin/wineserver` together with `wine` binaries from a different version, which explains the error.

**Comment by heit0r** [comment]
Interesting. Idk why I have this partial installation. Once I didn't wanted to the official packager to update wine so I downloaded wine (6.27 I think) myself, builded and flagged to do not update it in pacman.conf. But later I removed it and unflagged wine. Do you think I should remove wine and all these paths? What would you do? How did you recognize _a partial installation_?

Thank you.

**Comment by robbert-vdh** [comment]
Doing those things through your package manager won't fix anything. You'll need to remove the Wine files from `/usr/local` by hand. Ideally you wouldn't have anything in `/usr/local/{bin,lib}` at all. Software should be managed by your package manager, and if that's not possible it's almost always a better idea to keep it contained within your home directory.

**Comment by heit0r** [comment]
I did it :D and couldn't do without your help.

I uninstalled wine, winetricks and yabridge.
Removed .vst3 and .wine paths.
Removed /usr/local/bin/wineserver and /usr/local/bin/wineserver and anything I could identify as belonging to a previous wine installation.

Reinstalled wine-staging that now outputs as 7:
$ `wine --version`
wine-7.0-rc2 (Staging)
Reinstalled `yabridge`(I think there was no need to reinstall it)
Ran `yabridgctl sync` and Ardour6 loaded them all and apparently everything sounds fine.

Thank you so so much.

![2022-01-18_21-14](https://user-images.githubusercontent.com/38054654/150039533-f2542beb-0c4f-42b1-85ac-9a6a179cc1ff.png)

**Comment by heit0r** [comment]
I did it :D and couldn't do without your help.

I uninstalled `wine winetricks yabridge`
Removed .vst3 and .wine paths.
Removed /usr/local/bin/wineserver and /usr/local/bin/wineserver and anything I could identify as belonging to a previous wine installation.

Reinstalled wine-staging that now outputs as 7:
$ `wine --version`
wine-7.0-rc2 (Staging)


$ `which wine wine64 wine64-preloader wineserver`now outputs as
/usr/bin/wine
/usr/bin/wine64
/usr/bin/wine64-preloader
/usr/bin/wineserver


Reinstalled `yabridge`(I think there was no need to reinstall it)
Ran `yabridgctl sync` and Ardour6 loaded them all and apparently everything sounds fine.

Thank you so so much.

![2022-01-18_21-14](https://user-images.githubusercontent.com/38054654/150039764-dd6e3fe6-8e74-4789-adca-b3413ca009c2.png)
